plugin.rb

```diff
@@ -137,6 +137,7 @@ def self.accept_answer!(post, acting_user, topic: nil)
 
     def self.unaccept_answer!(post, topic: nil)
       topic ||= post.topic
+      topic ||= Topic.unscoped.find_by(id: post.topic_id)
```

Comment:
Then we won't be able to "delete" users with solved posts. Which is what this PR is trying to fix.

Reply:
I don't understand.

```ruby
topic ||= Topic.unscoped.find_by(id: post.topic_id)
```

if `topic` here is `nil` we will still fail at next line as we used to fail before, because `topic.id` will fail:

```
DistributedMutex.synchronize("discourse_solved_toggle_answer_#{topic.id}") do
```

Your change is fixing the fact that now we will find the topic in this case, so most likely we will never fail this again, but what Im telling if that if for any reason `post.topic_id` doesn’t actually exist, if we use `find` we will have the reason, instead of having a weird error later in hojack.